### PR TITLE
updated to KNP

### PR DESCRIPTION
**Pull request by wadeking98**
@esune @WadeBarnes

### CODE REVIEW

File: openshift/templates/iiw-book-agent/iiw-book-agent-deploy.yaml [hunk 1]
```diff
@@ -4,48 +4,102 @@ apiVersion: v1
...
-        - - role=service
-          - app=${APP_GROUP}
-          - env=${TAG_NAME}
-          - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
+      description: Allow the service to access the agent.
+      ingress:
+        - from:
+            - podSelector:
+                matchLabels:
+                  role: service
```
Comment (esune) [inline openshift/templates/iiw-book-agent/iiw-book-agent-deploy.yaml]: The role of the service can be changed at deployment time by updating the parameters: this should also be populated by a parameter, in order to make it possible to update it if necessary.

File: openshift/templates/iiw-book-service/iiw-book-service-deploy.yaml [hunk 2]
```diff
@@ -4,50 +4,104 @@ apiVersion: v1
...
-      - - role=agent
-        - app=${APP_GROUP}
-        - env=${TAG_NAME}
-        - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
+    description: Allow the agent to access the service.
+    ingress:
+      - from:
+        - podSelector:
+            matchLabels:
+              role: agent
```
Comment (esune) [inline openshift/templates/iiw-book-service/iiw-book-service-deploy.yaml]: The role of the agent can be changed at deployment time by updating the parameters: this should also be populated by a parameter, in order to make it possible to update it if necessary.

File: openshift/templates/nsp/deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  #     source:
+  #       - - $namespace=${NAMESPACE_NAME}-${ENV_NAME}
+  #     destination:
+  #       - - ext:name=internet-build-resources
+
+parameters:
+  - name: NAMESPACE_NAME
+    displayName: The target namespace for the resources.
+    required: true
+    value: 01a527
```
Comment (esune) [inline openshift/templates/nsp/deploy.yaml]: This should be updated to the namespace targeted by these services

File: openshift/templates/postgresql/postgresql-deploy.yaml [hunk 4]
```diff
@@ -6,6 +6,65 @@ metadata:
...
+        app: ${APP_NAME}
+        env: ${TAG_NAME}
+    spec:
+      description: |
+        Allow the api(s) to access the db.
+      ingress:
+        - from:
+            - podSelector:
+                matchLabels:
+                  role: service
```
Comment (esune) [inline openshift/templates/postgresql/postgresql-deploy.yaml]: The role of the service can be changed at deployment time by updating the parameters: this should also be populated by a parameter, in order to make it possible to update it if necessary.

**Review by esune — CHANGES_REQUESTED** [review]

**Comment by esune** [comment]
The global `build` NSP is also missing, please add it to the repository.

File: openshift/templates/iiw-book-agent/iiw-book-agent-deploy.yaml (hunk 1 above)
Comment (wadeking98) [reply openshift/templates/iiw-book-agent/iiw-book-agent-deploy.yaml]: change completed

File: openshift/templates/nsp/deploy.yaml (hunk 3 above)
Comment (wadeking98) [reply openshift/templates/nsp/deploy.yaml]: change completed

File: openshift/templates/nsp/build.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        name: pods-to-external-network
+        env: ${ENV_NAME}
+    spec:
+      description: |
+        Allow the builds to access the internet.
+        This only needs to be specified once per environment.
+      source:
+        - - $namespace=${NAMESPACE_NAME}-${ENV_NAME}
+      destination:
+        - - ext:name=internet-build-resources
```
Comment (esune) [inline openshift/templates/nsp/build.yaml]: OUtadet, needs to be updated with new KNPs

**Review by esune — CHANGES_REQUESTED** [review]

File: openshift/templates/nsp/build.yaml (hunk 5 above)
Comment (wadeking98) [reply openshift/templates/nsp/build.yaml]: updated nsp

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
I recommend renaming the `postgresql` template folders and files to `db`, to complete the migration from implementation specific naming conventions.

**Review by WadeBarnes — APPROVED** [review]

**Review by esune — APPROVED** [review]

**Pull request merged**